src/main/java/com/github/quadflask/smartcrop/SmartCrop.java

```diff
@@ -9,284 +9,375 @@
-	private float thirds(float x) {
-		x = ((x - (1 / 3f) + 1.0f) % 2.0f * 0.5f - 0.5f) * 16f;
-		return Math.max(1.0f - x * x, 0);
-	}
+    private Options options;
+    private int[] cd;
+
+    public SmartCrop() {
+        this(Options.DEFAULT);
+    }
+
+    public SmartCrop(Options options) {
+        this.options = options;
+    }
+
+    public static CropResult analyze(Options options, BufferedImage input) {
+        return new SmartCrop(options).analyze(input);
+    }
+
+    public CropResult analyze(BufferedImage input) {
+        Image inputI = new Image(input);
+
+
+        Image outputI = new Image(input.getWidth(), input.getHeight());
+
+        prepareCie(inputI);
+        edgeDetect(inputI, outputI);
+        skinDetect(inputI, outputI);
+        saturationDetect(inputI, outputI);
+        applyBoosts(outputI);
+
+        BufferedImage output = new BufferedImage(input.getWidth(), input.getHeight(), options.getBufferedBitmapType());
+        output.setRGB(0, 0, input.getWidth(), input.getHeight(), outputI.getRGB(), 0, input.getWidth());
+
+        BufferedImage score = new BufferedImage(input.getWidth() / options.getScoreDownSample(), input.getHeight() / options.getScoreDownSample(), options.getBufferedBitmapType());
+        score.getGraphics().drawImage(output, 0, 0, score.getWidth(), score.getHeight(), 0, 0, output.getWidth(), output.getHeight(), null);
+        Image scoreI = new Image(score);
+
+
+        float topScore = Float.NEGATIVE_INFINITY;
+        Crop topCrop = null;
+        List<Crop> crops = crops(scoreI);
+
+        for (Crop crop : crops) {
+            crop.score = score(scoreI, crop);
+            if (crop.score.total > topScore) {
+                topCrop = crop;
+                topScore = crop.score.total;
+            }
+
+
+            crop.x *= options.getScoreDownSample();
+            crop.y *= options.getScoreDownSample();
+            crop.width *= options.getScoreDownSample();
+            crop.height *= options.getScoreDownSample();
+
+
+            if (options.getBoost() != null && options.getBoost().length == 1) {
+                Crop r = options.getBoost()[0];
+
+                if (crop.x > r.x) {
+                    crop.x = r.x;
+                }
+
+                if (crop.width < r.width) {
+                    crop.width = r.width;
+                }
+
+                if (crop.y > r.y) {
+                    crop.y = r.y;
+                } else if ((crop.y + crop.height) < (r.y + r.height)) {
+                    crop.height = crop.height + (r.y + r.height - crop.y - crop.height) * options.getScoreDownSample();
+                }
+
+                if (crop.height < r.height) {
+                    crop.y = r.height;
+                }
+            }
+
+
+        }
+
+        CropResult result = CropResult.newInstance(topCrop, crops, output, createCrop(input, topCrop));
+        if (options.isDebug()) {
+            Graphics graphics = output.getGraphics();
```

Comment:
As you are updating the alpha channel due to the boost calculation, you might want to drop the alpha channel from debug output (otherwise all pixels where the alpha channel is 0 won't be visible). I use the following to create a new BufferedImage without the alpha channel.

    BufferedImage debugOutput = new BufferedImage(output.getWidth(), output.getHeight(), BufferedImage.TYPE_INT_RGB);
    // Drop alpha channel from debug output
    BandCombineOp filterAlpha = new BandCombineOp(
        // RGBA -> RGB
        new float[][] {
            {1.0f, 0.0f, 0.0f, 0.0f},
            {0.0f, 1.0f, 0.0f, 0.0f},
            {0.0f, 0.0f, 1.0f, 0.0f}
        }, null
    );
    filterAlpha.filter(output.getRaster(), debugOutput.getRaster());